### ISSUE TITLE
Add support for CMIP6 data in Pangeo's Google Cloud Storage 

### ISSUE THREAD

**Issue by andersy005** [NCAR/intake-esm#132]
> How can we make intake-esm work with our cloud datasets?

One way to achieve this is to replicate what I did for the AWS CESM-LENS data. The YAML file used to build the catalog looks like this: https://github.com/NCAR/intake-esm-datastore/blob/master/collection-input/aws-cesm1-le-collection.yml

```
name: AWS-CESM1-LE
collection_type: cesm-aws
data_sources:
  land:
    locations:
      - name: monthly
        loc_type: aws-s3
        direct_access: True
        urlpath: s3://ncar-cesm-lens/lnd/monthly
        file_extension: .zarr
      - name: daily
        loc_type: aws-s3
        direct_access: True 
        urlpath: s3://ncar-cesm-lens/lnd/daily
        file_extension: .zarr 

  ocean:
    locations:
      - name: monthly
        loc_type: aws-s3
        direct_access: True
        urlpath: s3://ncar-cesm-lens/ocn/monthly
        file_extension: .zarr
...
```

For CMIP6 data in the cloud, we could have something like:

```
name: GCS-CMIP6
collection_type: cmip6-gcs
data_sources:
  3hr:
    locations:
      - name: historical 3hourly atmosphere data
        loc_type: gcs
        direct_access: True
        urlpath: gcs://pangeo-data/CMIP6_3hr
        file_extension: .zarr
  monthly:
     locations:
       - name: historical monthly ocean data
         loc_type: gcs
         direct_access: True
         urlpath: gcs://pangeo-data/CMIP6_Omon
         file_extension: .zarr
    
...
```

**Comment by andersy005** [comment]
> @AndersonBanihirwe, thanks for this - very helpful.  As we see in your demo notebook, intake.open_esm_metadatastore can generate the catalog on the fly (so that it includes all of the currently available data).  The intake.open_catalog method accesses pre-made catalogs, which can then contain extra info, including who uploaded the data, etc.  Either method then allows the user to search for the same dataset (whether a zarr store or netcdf files) by identifying the CMIP6 source_id, table_id, variable_id, etc.  and automatically loading as an xarray dataset.  Do I understand this correctly?

> my zarr stores have a slightly different directory structure - since I am concatenating the netcdf files (sometimes from different version directories), the zarr stores are located at
```
        <mip_era>/
            <activity_id>/
                <institution_id>/
                    <source_id>/
                        <experiment_id>/
                            <member_id>/
                                <table_id>/
                                    <variable_id>/
                                        <grid_label>
```
So I can't use the standard CMIP6Collection class in cmip.py - if you could also help me with that, I would appreciate

**Comment by andersy005** [comment]
@naomi-henderson, 

> So I can't use the standard CMIP6Collection class in cmip.py

I am thinking of implementing another class for the CMIP6 data in the cloud due to the differences in how they are constructed and how they are loaded in xarray. 

I will open a PR and I will ping you when it's ready

**Comment by naomi-henderson** [comment]
Yes, this is a much better place to have this discussion - thanks!

**Comment by rabernat** [comment]
> I am thinking of implementing another class for the CMIP6 data in the cloud due to the differences in how they are constructed and how they are loaded in xarray.

I would be interested in understanding more detail about what the differences are. Could someone summarize?

The way we have uploaded things is was chosen somewhat randomly. We are experimenting here. If there is some established convention that we think would be more effective, we can change how the data are stored in GCS.

**Comment by n-henderson** [comment]
@rabernat , it is the ability of `intake-esm` to automatically generate an intake catalog from a specific directory structure that requires a different class.  If there were a one-to-one correspondence between zarr stores and netcdf files, there might be an easier work around. But since we are concatenating the netcdf files in time, and this often combines files from multiple 'version' subdirectories, the directory structure/ naming structure cannot be the same (i.e., there is one directory level less, no 'version' directory, thus no 'version' key).  The other keys, i.e. 'source_id', 'table_id', etc can still be inferred from the directory structure in an identical fashion in both cases.  So it is nothing to do with how we are storing the data in GCS, in fact I can't use  the current version of `intake-esm` to catalog the zarr stores even on my local linux machine.  

@andersy005 , anything to add?

**Comment by rabernat** [comment]
Ok thanks so much for the clarification! I think I understand now.

Rather than jumping into more coding, the right thing to do from a design point of view would be to formally describe the two conventions via a specification / design document. This doesn't have to be painful. This could help us decide how to refactor intake-esm to accommodate both cases, as well as future ones we that we will think of in the course of writing the specification.

Once such a document exists, the coding should be relatively straightforward. @andersy005 - perhaps it already exists in your mind! If so, it would be great to get it out in the open.

**Comment by andersy005** [comment]
Fixed in #135
